src/test/java/duke/data/TaskListTest.java

```diff
@@ -0,0 +1,22 @@
+package duke.data;
+
+import duke.commands.Task;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class TaskListTest {
+    @Test
+    public void testDelete() {
+        ArrayList<Task> test = new ArrayList<>();
+        Task task = new Task("test");
+        test.add(task);
+        test.add(task);
+        test.add(task);
+        TaskList testList = new TaskList(test);
+        testList.delete(1);
+        assertEquals(2, testList.size());
+    }
```

Comment:
I feel that the test method name can be more descriptive. Perhaps `taskDeletion_nonEmptyTaskList_successful`? Similarly for the other tests as well